src/config_file.c

```diff
@@ -239,8 +239,49 @@ gboolean load_config(const gchar *filename, RaucConfig **config, GError **error)
 		}
 	}
 
-	c->boot_default_attempts = key_file_consume_integer(key_file, "system", "boot-attempts", NULL);
-	c->boot_attempts_primary = key_file_consume_integer(key_file, "system", "boot-attempts-primary", NULL);
+	c->boot_default_attempts = key_file_consume_integer(key_file, "system", "boot-attempts", &ierror);
+	if (g_error_matches(ierror, G_KEY_FILE_ERROR, G_KEY_FILE_ERROR_KEY_NOT_FOUND)) {
+		c->boot_default_attempts = -1;
+		g_clear_error(&ierror);
+	} else if (ierror) {
+		g_propagate_error(error, ierror);
+		return FALSE;
+	}
+	if (c->boot_default_attempts < -1) {
+		g_set_error(
+				error,
+				R_CONFIG_ERROR,
+				R_CONFIG_ERROR_BOOTLOADER,
+				"Value for \"boot-attempts\" must not be negative");
+		return FALSE;
```

Comment:
This would be confusing error message if the key was missing. The same applies below. We should probably set a real default value.

Reply:
If the key isn't found, we set `-1` and we test for `< -1` to emit the error message, thus this should be safe.
It might only have an 'unexpected' result if someone writes `-1` in the config file :smirk:.

I chose this approach since the default value is currently set by bootchooser.c for each backend individually (based on a `#define`).

Reply:
Hmm, I read this several times and always saw `<= -1`. Perhaps use 0 as "unspecified"?

Reply:
I was not sure if there could be any valid use case or test setup that should allow setting 0 from config. But probably enforcing reset attempts to be > 0 is ok here.

Reply:
Otherwise, I wouldn't mind moving the default to the config file parsing.